packages/plugins/minos-router-graphql/minos/plugins/graphql/builders/schema.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import (
+    annotations,
+)
+
+from functools import (
+    wraps,
+)
+from inspect import (
+    isawaitable,
+)
+from typing import (
+    Any,
+)
+
+from graphql import (
+    GraphQLArgument,
+    GraphQLField,
+    GraphQLObjectType,
+    GraphQLSchema,
+    GraphQLString,
+)
+
+from minos.networks import (
+    EnrouteDecoratorKind,
+    InMemoryRequest,
+)
+
+
+class GraphQLSchemaBuilder:
+    """TODO"""
+
+    def __init__(self, *args, **kwargs):
+        self.schema = GraphQLSchema(**kwargs)
+
+    @classmethod
+    def build(cls, routes) -> GraphQLSchema:
+        """TODO"""
+        schema_args = cls._build(routes)
+        return cls(**schema_args).schema
+
+    @classmethod
+    def _build(cls, routes) -> dict:
+        query = cls._build_queries(routes)
+        mutation = cls._build_mutations(routes)
+
+        return {"query": query, "mutation": mutation}
+
+    @staticmethod
+    def adapt_callback(callback):
+        """TODO"""
+
+        @wraps(callback)
+        async def _wrapper(_source, _info, request: Any = None):
+            request = InMemoryRequest(request)
+
+            response = callback(request)
+            if isawaitable(response):
+                response = await response
+
+            return await response.content()
+
+        return _wrapper
+
+    @classmethod
+    def _build_queries(cls, routes):
+        fields = dict()
+        for route, callback in routes.items():
+            callback = cls.adapt_callback(callback)
+            if route.KIND == EnrouteDecoratorKind.Query:
+                fields[route.name] = cls._build_field(route, callback)
+
+        result = GraphQLObjectType(
+            "Query", fields={"hello": GraphQLField(GraphQLString, resolve=lambda obj, info: "world")}
+        )
```

Comment:
We should try to find a different solution for this problem. 😅

Reply:
I have simplified it, but it is still necessary to pass a base structure. The good thing is that this is transparent to the user.
```python
result = GraphQLObjectType(
            "Query", fields={"dummy": GraphQLString}
        )
```

https://stackoverflow.com/questions/54322029/graphqljs-query-root-type-must-be-provided

Reply:
You were right @vladyslav-fenchak , I do not understand why is that constraint but we need to follow it hahaha